packages/cozy-harvest/src/Manifest.js

```diff
@@ -0,0 +1,63 @@
+export const legacyLoginFields = [
+  'login',
+  'identifier',
+  'new_identifier',
+  'email'
+]
+
+export const roleShouldBeRequired = ['identifier', 'password']
+
+export const legacyEncryptedFields = [
+  'secret',
+  'dob',
+  'code',
+  'answer',
+  'access_token',
+  'refresh_token',
+  'appSecret'
+]
+
+export const Manifest = {
+  sanitize: manifest => {
+    if (!manifest.fields) return manifest
+    manifest = Manifest.sanitizeIdentifier(manifest)
+    manifest = Manifest.sanitizeIsRequired(manifest)
+    manifest = Manifest.sanitizeEncrypted(manifest)
+    return manifest
+  },
+
+  sanitizeIdentifier: manifest => {
+    for (let fieldName in manifest.fields)
+      if (manifest.fields[fieldName].role === 'identifier') return manifest
+
+    for (let name of legacyLoginFields)
+      if (manifest.fields[name]) {
+        manifest.fields[name].role = 'identifier'
+        return manifest
+      }
+
+    for (let fieldName in manifest.fields)
+      if (fieldName !== 'password')
+        manifest.fields[fieldName].role = 'identifier'
```

Comment:
If several fields are not passwords, all will be marked as `identifiers`.

This should need and additional test imho.